Protecs/src/main/java/uk/ac/ucl/protecs/behaviours/MovementBehaviourFramework.java

```diff
@@ -72,49 +72,114 @@ public double next(Steppable s, double time) {
 					return myWorld.params.ticks_per_day; // rest until tomorrow
 
-				
-				// if it's morning, go out for the day
+				// if it's morning, go out for the day, reset the number of contacts they will have
 				if(hour >= myWorld.params.hour_start_day_weekday){ 
+					// reset occurs at hour 2
+					p.resetWorkplaceContacts();
 
-					Location target;
-					target = myWorld.params.getTargetMoveAdminZone(p, day, myWorld.random.nextDouble(), myWorld.lockedDown);
-					// define workday
-					boolean goToWork = (p.isSchoolGoer() || target == p.getCommunityLocation()) // schoolgoer or going to own admin zone
-							&& myWorld.params.isWeekday(day);
+					return determineDailyRoutine(p, hour, day);
+				}
+				
+				return 1; // otherwise it's not the morning - stay home for now, but check in again later!
+			}
+			
+			private double determineDailyRoutine(Person p, int hour, int day) {
+				Location target;
+				target = myWorld.params.getTargetMoveAdminZone(p, day, myWorld.random.nextDouble(), myWorld.lockedDown);
+				// then check if they are supposed to leave the admin zone they are currently in. If so, then they cannot go to work.
+				boolean stayingInHomeDistrict = target.getId().equals(p.getHousehold().getRootSuperLocation().getId());
+				// First check if they are visiting another district
+				if (!stayingInHomeDistrict) p.setVisiting(true);
+				if (p.visitingNow() & !stayingInHomeDistrict) {
+					myWorld.outbound_trip_counter ++;
+					// travelling to another district!
+					p.transferTo(target);
+					p.setActivityNode(communityNode);
+					p.setAtWork(false);
+					assert ! p.getHousehold().getSuper().equals(target) : 
+						"set to travel to a different district but didn't, home/target " + p.getHousehold().getSuper().getId() + " " + target.getId();
+					 // stay out until time to go home!
+					return myWorld.params.hour_end_day_otherday - hour;
+				}
+				// if they aren't visiting, are they at work or the community?
+				else {
+					// Check they are going to work
+					myWorld.stay_home_counter ++;
+					boolean goToWork = myWorld.random.nextDouble() < myWorld.params.prob_go_to_work;
 
+					// if unemployed or homemaker don't go to work
+					if (p.isUnemployed()) {
+						goToWork = false;
+					}
+					// first check if there is any constraints to this occupations movements. Note that if the model is using this code block then 
+					// this person has not been immobilised and if their movement is constrained it will mean that they only go to the community and not to work
+					boolean movementConstrained = myWorld.params.OccupationConstraintList.containsKey(p.getEconStatus());
+						
 					if(myWorld.params.setting_perfectMixing) // in perfect mixing, just go to the community!
 						goToWork = false;
 					
+					if (goToWork & !movementConstrained) target = p.getWorkLocation();
+
+					
 					p.transferTo(target);
 					assert (p.getLocation().equals(target)) : "Transfer to target didn't work";
```

Comment:
general principle: I like to include the specific details of the offending variable combo, so if it's one particular person agent screwing up, we know they're the problem; if it's one particular location, that's the issue, etc. @RobertManningSmith is adding a new issue about error message verbosity!!